src/plugins/vis_type_timeseries/public/application/components/index_pattern_suggest.tsx

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React, { useEffect, useState, useCallback } from 'react';
+import { EuiSuggest, EuiSuggestProps } from '@elastic/eui';
+import { getDataStart } from '../../services';
+import { INDEXES_SEPARATOR } from '../../../common/constants';
+
+interface IndexPatternSuggestProps {
+  value: string;
+  indexPatternName: string;
+  defaultIndexPattern: string;
+  onChange: Function;
+  disabled?: boolean;
+}
+
+const toSuggesOptions = (options: string[]) =>
+  options.map((label) => ({ type: { iconType: 'indexPatternApp', color: 'tint5' }, label }));
+
+export const IndexPatternSuggest = ({
+  value,
+  indexPatternName,
+  onChange,
+  disabled,
+  defaultIndexPattern,
+}: IndexPatternSuggestProps) => {
+  const [availableIndexes, setAvailableIndexes] = useState<string[]>([]);
+  const [inputValue, setInputValue] = useState<string>(value);
+
+  const splittedIndexes = value.split(INDEXES_SEPARATOR);
+  const prefix = splittedIndexes[Math.max(0, splittedIndexes.length - 1)];
+
+  const suggestions = availableIndexes.filter(
+    (index) =>
+      index !== prefix &&
+      index.startsWith(prefix) &&
```

Comment:
I suggest to not return suggestions that `startWith` the input value but to return suggestions that contain the input value. In that way, I have to remember only one word from my index pattern and not how it starts. For example I write `logs` and the `kibana_sampl_data_logs` index pattern is suggested. I think that this will improve the UX

Reply:
changed that and I like how it works now ❤️